### ISSUE TITLE
Atomicapp image fails with 1.7.0 : client and server don't have same version (client : 1.19, server: 1.18)

### ISSUE THREAD

**Issue by sub-mod** [projectatomic/atomicapp#174]
If you have created a atomicapp image recently .
It will download docker 1.7.0 and it gives the below error.
I have tried to overcome that issue by putting docker-1.6.0 in yum command  for now.
@goern @aweiteka @vpavlin 

 /home/development/atomicapp-run/atomicapp.sh -v --answers=/home/development/atomic_jsonbrooks/nulecule/examples/tmp/answers.conf run submod/guestbookphp-app
2015-07-15 14:19:19,496 - atomicapp.install - INFO - App name is submod/guestbookphp-app, will be populated to /atomicapp
2015-07-15 14:19:19,496 - atomicapp.utils - INFO - atomicapp.status.info.message=Loading app submod/guestbookphp-app .
2015-07-15 14:19:19,497 - atomicapp.utils - INFO - atomicapp.status.answer.message={"general": {"namespace": "default", "provider": "kubernetes"}}
2015-07-15 14:19:19,497 - atomicapp.nulecule_base - DEBUG - {'namespace': 'default', 'provider': 'kubernetes'}
Error response from daemon: client and server don't have same version (client : 1.19, server: 1.18)
Traceback (most recent call last):
  File "/usr/bin/atomicapp", line 9, in <module>
    load_entry_point('atomicapp==0.1.9', 'console_scripts', 'atomicapp')()
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/cli/main.py", line 101, in main
    cli.run()
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/cli/main.py", line 82, in run
    args.func(args)
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/cli/main.py", line 22, in cli_run
    ae = Run(**vars(args))
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/run.py", line 63, in __init__
    install.install()
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/install.py", line 99, in install
    self.nulecule_base.pullApp()
  File "/usr/lib/python2.7/site-packages/atomicapp-0.1.9-py2.7.egg/atomicapp/nulecule_base.py", line 318, in pullApp
    image_id = subprocess.check_output(check_cmd)
  File "/usr/lib64/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['docker', 'images', '-q', 'submod/guestbookphp-app']' returned non-zero exit status 1
Needed for https://github.com/cockpit-project/cockpit/pull/2474

**Comment by vpavlin** [comment]
The simple fix would be to not only mount `docker.sock` but also `/usr/bin/docker`

**Comment by vpavlin** [comment]
Or...
```
[root@localhost bbb]# docker version
Client version: 1.7.0.fc22
Client API version: 1.19
Package Version (client): docker-1.7.0-8.gita2cd872.fc22.x86_64
Go version (client): go1.4.2
Git commit (client): a2cd872/1.7.0
OS/Arch (client): linux/amd64
Error response from daemon: client and server don't have same version (client : 1.19, server: 1.18)
[root@localhost bbb]# /host/usr/bin/docker version
Client version: 1.5.0-dev
Client API version: 1.18
Go version (client): go1.3.3
Git commit (client): fc0329b/1.5.0
OS/Arch (client): linux/amd64
Server version: 1.5.0-dev
Server API version: 1.18
Go version (server): go1.3.3
Git commit (server): fc0329b/1.5.0
OS/Arch (server): linux/amd64
```

we can use `/host/usr/bin/docker` instead

**Comment by aweiteka** [comment]
@vpavlin that approach would be consistent with the other providers. Is it the safest approach?

**Comment by goern** [comment]
+1 for /host/usr/bin/docker, lets consider it a feature: Atomic App provider use the binary interface provided by the host on which `atomic {install|run}` is executed

**Comment by vpavlin** [comment]
It's not only about provider. This is more about the "packaging" - the problem raises when Atomic App tries to unpack external dependencies. But anyway - as long as we rely on `/host` being available it's safe.

**Comment by ggallen** [comment]
I see the same problem with the latest rpms on CentOS.

[root@dhcp-41-86 atomic-test]# rpm -qa | grep kub
kubernetes-node-1.0.0-0.8.gitb2dafda.el7.x86_64
kubernetes-master-1.0.0-0.8.gitb2dafda.el7.x86_64
kubernetes-1.0.0-0.8.gitb2dafda.el7.x86_64
[root@dhcp-41-86 atomic-test]# rpm -qa | grep atom
atomicapp-0.1.1-1.el7.noarch
atomic-1.0-108.el7.centos.x86_64
[root@dhcp-41-86 atomic-test]# rpm -qa | grep docker
docker-python-1.4.0-108.el7.centos.x86_64
docker-selinux-1.7.1-108.el7.centos.x86_64
docker-1.7.1-108.el7.centos.x86_64
[root@dhcp-41-86 atomic-test]#

**Comment by dustymabe** [comment]
I don't know if using the host `docker` is going to be sufficient. I tried to do this by adding a `RUN ln -s /host/usr/bin/docker /usr/bin/docker` in the Dockerfile deleting docker from the list of rpms that gets included. I end up with this inside the container:

```
[root@f22 atomicapp]# ldd /usr/bin/docker
        linux-vdso.so.1 =>  (0x00007fff7f5e2000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fd9c1420000)
        libsqlite3.so.0 => /lib64/libsqlite3.so.0 (0x00007fd9c116b000)
        libdevmapper.so.1.02 => not found
        libc.so.6 => /lib64/libc.so.6 (0x00007fd9c0daa000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fd9c163c000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fd9c0ba6000)
```

I worked around this by modifying `LD_LIBRARY_PATH`:

```
[root@f22 atomicapp]# export LD_LIBRARY_PATH="/lib64/:/lib/:/host/lib64/:/host/lib/"
[root@f22 atomicapp]# ldd /usr/bin/docker
        linux-vdso.so.1 =>  (0x00007ffd2f1fb000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fe6beafc000)
        libsqlite3.so.0 => /lib64/libsqlite3.so.0 (0x00007fe6be847000)
        libdevmapper.so.1.02 => /host/lib64/libdevmapper.so.1.02 (0x00007fe6be5fd000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fe6be23c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fe6bed18000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fe6be038000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007fe6bde13000)
        libsepol.so.1 => /lib64/libsepol.so.1 (0x00007fe6bdbce000)
        libudev.so.1 => /lib64/libudev.so.1 (0x00007fe6bd9bc000)
        libpcre.so.1 => /lib64/libpcre.so.1 (0x00007fe6bd75b000)
        liblzma.so.5 => /lib64/liblzma.so.5 (0x00007fe6bd536000)
        librt.so.1 => /lib64/librt.so.1 (0x00007fe6bd32e000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fe6bd118000)
```

and.. it works. This is on a F22 host with a centos based atomicapp container. I don't know if this is something we can use or not.

**Comment by goern** [comment]
proposed solution to this issue it to let atomicapp execution gracefully die and make a 'please update' statement to the user/deployment.

**Comment by dustymabe** [comment]
Fixed by #361.. Marking as done and will close when beta3 branch is merged.
